modules/core/src/directx.cpp

```diff
@@ -234,12 +234,116 @@ int getTypeFromD3DFORMAT(const int iD3DFORMAT)
+    clEnqueueAcquireD3D11ObjectsNV_fn clEnqueueAcquireD3D11ObjectsNV = NULL;
+    clEnqueueReleaseD3D11ObjectsNV_fn clEnqueueReleaseD3D11ObjectsNV = NULL;
+#endif
+    clCreateFromD3D11Texture2DKHR_fn clCreateFromD3D11Texture2DKHR = NULL;
+    clEnqueueAcquireD3D11ObjectsKHR_fn clEnqueueAcquireD3D11ObjectsKHR = NULL;
+    clEnqueueReleaseD3D11ObjectsKHR_fn clEnqueueReleaseD3D11ObjectsKHR = NULL;
+
+    clCreateFromD3D10Texture2DKHR_fn clCreateFromD3D10Texture2DKHR = NULL;
+    clEnqueueAcquireD3D10ObjectsKHR_fn clEnqueueAcquireD3D10ObjectsKHR = NULL;
+    clEnqueueReleaseD3D10ObjectsKHR_fn clEnqueueReleaseD3D10ObjectsKHR = NULL;
+
+    clCreateFromDX9MediaSurfaceKHR_fn clCreateFromDX9MediaSurfaceKHR = NULL;
+    clEnqueueAcquireDX9MediaSurfacesKHR_fn clEnqueueAcquireDX9MediaSurfacesKHR = NULL;
+    clEnqueueReleaseDX9MediaSurfacesKHR_fn clEnqueueReleaseDX9MediaSurfacesKHR = NULL;
+
+    cl_platform_id initializedPlatform9 = NULL;
+    cl_platform_id initializedPlatform10 = NULL;
+    cl_platform_id initializedPlatform11 = NULL;
+
+    cl_platform_id getPlatform()
+    {
+        if (!platform)
+        {
+            cl_device_id current_device = (cl_device_id)ocl::Device::getDefault().ptr();
+            CV_Assert(current_device);
+
+            cl_uint numPlatforms;
+            cl_int status = clGetPlatformIDs(0, NULL, &numPlatforms);
+            if (status != CL_SUCCESS)
+                CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get number of platforms");
+            if (numPlatforms == 0)
+                CV_Error(cv::Error::OpenCLInitError, "OpenCL: No available platforms");
+
+            std::vector<cl_platform_id> platforms(numPlatforms);
+            status = clGetPlatformIDs(numPlatforms, &platforms[0], NULL);
+            if (status != CL_SUCCESS)
+                CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get platforms");
+
+            bool found = false;
+            for (int i = 0; !found && i < (int)numPlatforms; i++)
+            {
+                cl_uint numDevices;
+                status = clGetDeviceIDs(platforms[i], CL_DEVICE_TYPE_ALL, 0, NULL, &numDevices);
+                if (status != CL_SUCCESS && status != CL_DEVICE_NOT_FOUND)
+                    CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get number of devices");
+                if (numDevices == 0)
+                    continue;
+
+                std::vector<cl_device_id> devices(numDevices);
+                status = clGetDeviceIDs(platforms[i], CL_DEVICE_TYPE_ALL, numDevices, &devices[0], 0);
+                if (status != CL_SUCCESS)
+                    CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get devices");
+
+                for (int j = 0; !found && j < (int)numDevices; j++)
+                {
+                    if (current_device == devices[j])
+                    {
+                        platform = platforms[i];
+                        found = true;
```

Comment:
Perhaps you want to use `clGetDeviceInfo()` call with `CL_DEVICE_PLATFORM`.

Reply:
Oh! Thank you!
Certainly the `platform` can be obtained using `clGetDeviceInfo()`. Change it to use this.